src/opustrainer/trainer.py

```diff
@@ -618,12 +618,12 @@ def run(self, *, batch_size:int=100) -> Iterable[List[str]]:
                 # Apply any modifiers to random lines in the batch, or sentence
                 # (Multiple modifiers could be applied to the same line!)
                 if self.stage.modifiers is not None:
-                    modifiers = self.stage.modifiers
+                    modifiers = self.stage.modifiers + self.curriculum.modifiers
```

Comment:
I don't like that this always adds the global modifiers at the end. Modifier order is important, and always adding the global ones at the end will either limit what you can put in the global ones, making it less useful, or surprise you.

I'd rather have people explicitly mix the two, e.g, support YAML like this:

```yaml
datasets:
  data: x.gz

stages:
  - stage1
  - stage2

modifiers: &global_mods
  - UpperCase: 0.1

stage1:
  mix:
    - data 1.0
    - until data 1

stage2:
  mix:
    - data 1.0
    - until data 1
  modifiers:
    - *global_mods
    - TitleCase: 0.1
```

This would also be easier to document: global modifiers are applied, unless the stage has its own modifiers. Then those override the global ones.

(Note: that yaml doesn't work yet because you end up with a list in a list, but we can safely flatten the modifier list to support this)

Reply:
yes, this would definitely be better.

Reply:
Should I add it, or are you working on it?

Reply:
I implemented it, just haven't pushed it yet. I was fixing the tests for it before I had to go down to go to the Friday talk